src/AppInstallerCommonCore/Public/winget/Experiment.h

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include <string>
+#include <type_traits>
+#include "AppInstallerStrings.h"
+
+namespace AppInstaller::Settings
+{
+    enum ExperimentToggleSource
+    {
+        Default = 0,
+        Policy,
+        UserSetting,
+    };
+
+    struct ExperimentState
+    {
+        ExperimentState() = default;
+        ExperimentState(bool isEnabled, ExperimentToggleSource toggleSource) : m_isEnabled(isEnabled), m_toggleSource(toggleSource) {}
+        bool IsEnabled() const { return m_isEnabled; }
+        ExperimentToggleSource ToggleSource() const { return m_toggleSource; }
+        std::string ToJson() const;
+    private:
+        ExperimentToggleSource m_toggleSource;
+        bool m_isEnabled;
+    };
+
+    struct Experiment

```

Comment:
Could we make experiments part of the experimental features?

Experimental features already have most of the things we may want: settings to be turned on/off, GPO to disable them wholesale, an API to check the state, a command to show the current state. The only thing we would need to add is the option that they may be set remotely for an experiment if there is no local override.

Reply:
I think we might still want a new GP for "allow experimental feature state to be changed remotely", but other than that it was my original thought that all experiments would likely have a feature.  I forget the reasons why I softened on that, but I do think that there may be some value to separating them as they are slightly different semantics.  One potential large reason is that if we use EFs exclusively, then a disabled GP for EFs also disables experiments.  Moving experiments out to their own thing means an admin can disable in-development features while still allowing for experiments.

Typically, an EF is work-in-progress and will eventually be released.  Using an experiment on an EF is more akin to controlled feature rollout.  I think it will probably happen that we hook that up so that an EF can be controlled by an experiment and thus we get CFR.

On the other hand, a true experiment is typically a small configuration or code change that is itself not risky from a local functionality perspective (the impact, such as changing the CDN *could* have major impacts on the overall system of course).  We might be testing something that we want to make permanent like where the DNS for the CDN points, or we might be checking to see if customer satisfaction numbers are higher if the default progress style is rainbow.

Reply:
> as they are slightly different semantics

If users are going to come across the two concepts, do we think the difference is clear enough for them to understand?

> an admin can disable in-development features while still allowing for experiments

Do we expect there will be admins wanting to do that?

Reply:
I don't think we want users to come across experiments really; the major reason for the control via settings is to enable *us* to test.

As to admins wanting to do that; I'm not sure.  But if we are not going to build any user facing interfaces around them, then I see the minor additional cost as acceptable.